src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -1400,10 +1406,11 @@ protected int getRequestThreadCount() {
         return session.getRequest().getDegreeOfConcurrency();
     }
 
-    protected Date getBuildStartTime() {
-        MavenExecutionRequest request = session.getRequest();
-        Date buildStartTime = request == null ? new Date() : request.getStartTime();
-        return buildStartTime == null ? new Date() : buildStartTime;
+    private Optional<Instant> getBuildStartTime() {
```

Comment:
uhm you're changing a protected method. this is breaking any backward compat.

Reply:
I might be missing something, but I don't expect that anyone extend from AbstractCompilerMojo externally, I have never seen maven plugins as public APIs that can be extended, in other words protected works for subclasses that are on a different package, so this should not break anything.

The only consumer is m-compiler-p itself, and even if this method is used in a subclass it should be package-protected (no modifier) since the subclasses CompilerMojo and TestCompilerMojo are on the same package.

Reply:
try a github search and you will see https://github.com/search?q=extends+AbstractCompilerMojo&type=code
Even if it's maybe a bad idea this was not private from start but it is what it is we have to live (almost forever0 with that.

Reply:
Actually there are many false-positive results, searching by  https://github.com/search?q=%22import+org.apache.maven.plugin.compiler.AbstractCompilerMojo%22&type=code gives a closer look of what is actually using this class, there are a few results, so yes, there are external consumers.

I will revert them to protected since it actually doesn't make a difference for what this PR is fixing.

BTW, forever is a long time, I expect this plugin and others to be rewritten someday and drop the legacy baggage ;-)